master/buildbot/process/buildrequestdistributor.py

```diff
@@ -1173,6 +1173,9 @@ def _selectNextBuildRequest(self, queue, asyncFunc):
 
     @defer.inlineCallbacks
     def _procesBuildRequestsActivityLoop(self):
```

Comment:
How frequently is this called?

Should we sleep for a while if master.changeServicesStateRunning is True?